src/native/libs/System.Security.Cryptography.Native.Android/net/dot/android/crypto/PalPbkdf2.java

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+package net.dot.android.crypto;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import javax.crypto.Mac;
+import javax.crypto.ShortBufferException;
+import javax.crypto.spec.SecretKeySpec;
+
+public final class PalPbkdf2 {
+    private static final int ERROR_UNSUPPORTED_ALGORITHM = -1;
+    private static final int SUCCESS = 1;
+
+    public static int pbkdf2OneShot(String algorithmName, byte[] password, byte[] salt, int iterations, byte[] destination)
```

Comment:
Since we control the Java API shape here, can we use NIO buffers here? That way we can avoid copying the salt into a Java array and remove another destination copy. (Can't avoid the password copy sadly with the existing Java API surface)

Reply:
I went ahead and updated the change to use a NIO buffer. Practically speaking, it makes no difference on the throughput of the function since there are only two buffers, and the length of them is small. Much has been said online about NIO buffers not being worth it unless you are dealing with "kilobyte" range of data and to just copy buffers. While a Kb is possible, it's far from the norm with PBKDF2.

If the (albeit small) increase in complexity is worth it to avoid two array allocations, happy to keep it. Otherwise I think using arrays keeps the code a little more neat.